access/api.go

```diff
@@ -52,6 +53,64 @@ type API interface {
 
 	GetExecutionResultForBlockID(ctx context.Context, blockID flow.Identifier) (*flow.ExecutionResult, error)
 	GetExecutionResultByID(ctx context.Context, id flow.Identifier) (*flow.ExecutionResult, error)
+
+	// SubscribeBlocks subscribes to the finalized or sealed blocks starting at the requested
+	// start block, up until the latest available block. Once the latest is
+	// reached, the stream will remain open and responses are sent for each new
+	// block as it becomes available.
+	//
+	// Each block is filtered by the provided block status, and only
+	// those blocks that match the status are returned.
+	//
+	// Only one of startBlockID or startHeight may be provided, Otherwise, an InvalidArgument error is returned.
+	// If neither startBlockID nor startHeight is provided, the latest sealed block is used.
+	//
+	// Parameters:
+	// - ctx: Context for the operation.
+	// - startBlockID: The identifier of the starting block. If provided, startHeight should be 0.
+	// - startHeight: The height of the starting block. If provided, startBlockID should be flow.ZeroID.
+	// - blockStatus: The status of the block, which could be only BlockStatusSealed or BlockStatusFinalized.
+	//
+	// If invalid parameters will be supplied SubscribeBlocks will return a failed subscription.
+	SubscribeBlocks(ctx context.Context, startBlockID flow.Identifier, startHeight uint64, blockStatus flow.BlockStatus) subscription.Subscription
```

Comment:
Let's also create new versions of the backend methods. this will remove the need for the special case handling of the start block